src/components/Dropdown/Dropdown.tsx

```diff
@@ -68,43 +68,22 @@ type PropsFromRedux = ReturnType<typeof mapStoreToProps> &
 const Dropdown: React.FC<PropsFromRedux> = ({ updateTaskData, allTasks }) => {
   const [open, setOpen] = useState<boolean>(false);
 
-  // use toggle for open and close dropdown menu when click on menu button
-  // changed approach to mouse hover
-  // const toggleMenu = () => {
-  //   setOpen(!open);
-  // };
-
-  // used useRef and useEffect for close dropdown menu
-  // when click other areas of app
-  // const handleClickOutside = (event: MouseEvent) => {
-  //   if (
-  //     container.current &&
-  //     !container.current.contains(event.target as Element)
-  //   ) {
-  //     setOpen(false);
-  //   }
-  // };
-
-  // useEffect(() => {
-  //   document.addEventListener("mousedown", handleClickOutside);
-  //   return () => document.removeEventListener("mousedown", handleClickOutside);
-  // }, []);
-
   const handleAddColumn = () => {
-    // setOpen(false);
     // store all current column ids in an array
     const currentColumnIds = Object.keys(allTasks.columns);
-    // get the last column id
-    const lastColumnId = currentColumnIds[currentColumnIds.length - 1];
-    // get the number of the last column id
-    const lastColumnIdNum = parseInt(
-      lastColumnId.charAt(lastColumnId.length - 1)
-    );
-    // set the new column id number to the last column id number + 1
-    // compared to the prior approach, which calculates the new column id
-    // based on the number of columns current exists which can duplicate column id
-    // e.x. delete one of the existing column and then add another column
-    const newColumnId = `column-${lastColumnIdNum + 1}`;
+    let newColumnId: string;
+    // if there are no columns exist
+    if (currentColumnIds.length === 0) {
+      // ste column id to column-1
+      newColumnId = "column-1";
+    } else {
+      // get the last column id
+      const lastColumnId = currentColumnIds[currentColumnIds.length - 1];
+      const regex = /^\D+/g; // find all leading non-digits
+      // replace all leading non-digits with empty string
+      const lastColumnIdNum = parseInt(lastColumnId.replace(regex, ""));
+      newColumnId = `column-${lastColumnIdNum + 1}`;
```

Comment:
I would recommend refactoring as much of this as you this into a shared function that tasks and columns can both use. Something like `getNewId` where you pass in the current ids (of the columns or tasks), and return the new ID.

Reply:
Added shared function getNewId, used in the dropdown and column component, and it seems to work.